earth2studio/data/jpss.py

```diff
@@ -0,0 +1,652 @@
+
+        # Filter fill values
+        if self._product_type in ["I", "M"]:
+            filter_threshold = 65529
+            filtered_data[filtered_data >= filter_threshold] = np.nan
+            filtered_data[filtered_data < 0] = np.nan  # Issue with data
+        elif self._product_type == "L2":
+            fill_value = dataset.attrs["_FillValue"].item()
+            filtered_data[filtered_data == fill_value] = np.nan
+
+        return filtered_data
+
+    @staticmethod
+    def _validate_satellite_and_product_type(satellite: str, product_type: str) -> None:
+        """Validate that requested satellite and product type are valid
+
+        Parameters
+        ----------
+        satellite : str
+            Satellite name to validate
+        product_type : str
+            Product type to validate
+
+        Raises
+        ------
+        ValueError
+            If satellite or product type is invalid
+        """
+        if satellite not in JPSS.VALID_SATELLITES:
+            raise ValueError(
+                f"Invalid satellite {satellite}. Must be one of {JPSS.VALID_SATELLITES}"
+            )
+        if product_type not in JPSS.VALID_PRODUCT_TYPES:
+            raise ValueError(
+                f"Invalid product_type {product_type}. Must be one of {JPSS.VALID_PRODUCT_TYPES}"
+            )
+
+    @staticmethod
+    def _validate_variables(variables: list[str], product_type: str = None) -> None:
+        """Validate that requested variables are compatible with the configured product type.
+
+        Parameters
+        ----------
+        variables : list[str]
+            List of variable names to validate
+        product_type : str, optional
+            Product type to validate against. If None, only checks if variables exist.
+
+        Raises
+        ------
+        ValueError
+            If variables have incompatible product types or invalid format
+        """
+        unknown_vars = []
+        non_product_vars = []
+
+        for var in variables:
+            # Check if variable exists in lexicon
+            if var not in JPSSLexicon.VOCAB:
+                unknown_vars.append(var)
+                continue
+
+            # Determine variable type if product_type is provided
+            if (
+                product_type is not None
+                and JPSSLexicon.get_item(var)[0] != product_type
+            ):
+                non_product_vars.append(var)
+
+        # Raise error for unknown variables
+        if unknown_vars:
+            raise ValueError(
+                f"Unknown VIIRS variables: {unknown_vars}. "
+                f"Please check the JPSS lexicon for available variables."
+            )
+
+        # Raise error for non-product variables
+        if non_product_vars:
+            raise ValueError(
+                f"Variables {non_product_vars} are incompatible with the configured product type {product_type}"
+            )
+
+    async def _get_s3_path(
```

Comment:
is this called for everything time stamp? How many files are in the bucket, im just curious if getting the possible files takes potentially a long time